CMakeLists.txt

```diff
@@ -1,35 +1,27 @@
-cmake_minimum_required(VERSION 2.4.4...3.15.0)
-set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS ON)
+# CMake 3.21 or later is required for PROJECT_IS_TOP_LEVEL
+cmake_minimum_required(VERSION 3.21...3.30)
```

Comment:
@madler to confirm, but if we have to be able conservative with CMake upgrades we could do:
```
if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
    set(PROJECT_IS_TOP_LEVEL TRUE)
endif()
```

and have the same `PROJECT_IS_TOP_LEVEL` 👍

Reply:
And what minimum Version do you want?

Reply:
I think it's very safe to deprecate anything below `3.0.0` (released [ten years ago](https://gitlab.kitware.com/cmake/cmake/-/tags?page=15)).

`3.10.0` is probably ["safe enough"](https://gitlab.kitware.com/cmake/cmake/-/tags?page=12) and added support to ARM on MSVC.

but of course it's really up to @madler to decide